src/writer.rs

```diff
@@ -844,6 +853,54 @@ mod write_vtk_impl {
             let buf = IOBuffer::from(data);
             self.write_buf::<BO>(buf)
         }
+        fn write_color_scalars_buf<BO: ByteOrder>(&mut self, buf: IOBuffer) -> Result {
+            fn write_buf_impl<T, W, F>(vec: Vec<T>, writer: &mut W, convert_to_u8: F) -> Result
+            where
+                W: WriteBytesExt,
+                F: Fn(T) -> u8,
+            {
+                for elem in vec {
+                    W::write_u8(writer, convert_to_u8(elem))?;
+                }
+                Ok(())
+            }
+            fn convert_float<T: Into<f64>>(input: T) -> u8 {
+                (input.into() * 255.0).round().clamp(0.0, 255.0) as u8
+            }
+
+            match buf {
```

Comment:
As your error description above describes we only dealt with f32 and u8 at the moment. Do you think we should accept all types right away or should we restrict it somehow and use the error above here? Same applies for the function implemented for the `AsciiWriter`.

Reply:
Good call! I was on the fence about whether the conversion should be lenient. I think most likely scenario is that users will create colour scalars with the right range of values, but may make an error in the type. I originally had it return the error, but then decided we might as well just convert since the conversion seemed like it would address the 99% of the errors anyways. I guess this kinda goes against the explicitness of most Rust code. What are your thoughts?

Reply:
Honestly we might as well handle all the cases if we have a look at this now. Though I am not sure if all cases are handled coherently yet. In general I think that at the end of the day we should be roundtrip safe. So binary -> ascii -> binary/ascii -> binary -> ascii should always lead to the same results. I'm not sure if that is the case yet. The tests I've mentioned in the other comment could be very helpful for that. I have added a couple more comments down below to explain what I mean.

Edit: Actually it is only the one `Bit` location where I think we don't need the multiplication. The other places look good. I'm interested to see if a test confirms that. Also if we handle all cases, the error above can be removed. 👍